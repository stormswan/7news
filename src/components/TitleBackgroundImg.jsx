import styled from 'styled-components'

const TitleBackgroundImg = styled.img.attrs ({
  src: props => props.src,
  alt: props => props.alt || 'title background image',
})`
  width:100%;
`

export default TitleBackgroundImg