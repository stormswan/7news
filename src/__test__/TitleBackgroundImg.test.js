import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import TitleBackgroundImg from '../components/TitleBackgroundImg'



test('Shapshot', () => {
  const titleBgImg = renderer.create( <TitleBackgroundImg/>).toJSON()
  expect(titleBgImg).toMatchSnapshot()
})


test('should work with default value', () => {
  const backgroundImg = renderer.create( <TitleBackgroundImg/>).toJSON()
  expect(backgroundImg).toHaveStyleRule('width','100%')
  expect(backgroundImg.props.alt).toBe('title background image')
})

test('should work with props value', () => {
  const alt= 'background'
  const src = '../images/tile.jpg'
  const backgroundImg = renderer.create( <TitleBackgroundImg alt={alt} src={src}/>).toJSON()
  expect(backgroundImg.props.alt).toBe(alt)
  expect(backgroundImg.props.src).toBe(src)
})