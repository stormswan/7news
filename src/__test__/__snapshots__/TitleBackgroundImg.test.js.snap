// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapshot 1`] = `
.c0 {
  width: 100%;
}

<img
  alt="title background image"
  className="c0"
  src={undefined}
/>
`;
